### PR TITLE
Defragmentation of NFA

### PR DESCRIPTION
**Pull request by martinhruska**
This PR:

- Implements defragmentation of NFA,
- Therefore needed renaming function for number predicate.

### CODE REVIEW

File: include/mata/number-predicate.hh [hunk 1]
```diff
@@ -327,6 +327,53 @@ namespace Mata {
...
+            /**
+             * Renames numbers in predicate according to the given @renaming. If a number is not present in @renaming
+             * it is renamed to base + offset value. Rationalization for this is that base should e.g.,
+             * higher than number of states in Nfa delta so we rename the initial or final states not presented in delta
+             * to numbers just after delta. Offset is then increased after encountering each of such states
+             * not presented in delta.
+             */
+            void rename(const std::vector<Number>& renaming, const Number base = 0) {
+                if (renaming.empty())
+                    return; // nothing to rename
```
Comment (Adda0) [inline include/mata/number-predicate.hh:340]: Do we want to rename the states in the number predicate with `base + offset` even if `renaming` is empty?

**Review by Adda0 — APPROVED** [review]
I approve the PR. The changes are reasonable.

File: include/mata/number-predicate.hh (hunk 1 above)
Comment (Adda0) [reply include/mata/number-predicate.hh:340]: We will discuss this in our meeting.

**Comment by martinhruska** [comment]
@tfiedor could you submit your review?

**Review by tfiedor — APPROVED** [review]
Looks fine, sorry forgot to submit the review.

**Pull request merged**